benchmarks/gluon_bench/benchmarks/parallel_seqset.go

```diff
@@ -45,23 +45,43 @@ func NewParallelSeqSetRandom(count uint32, numWorkers uint, generateIntervals, r
 	for i := uint(0); i < numWorkers; i++ {
 		list := make([]*imap.SeqSet, 0, count)
 
-		if generateIntervals && randomDrain {
-			panic("This is not yet supported")
-		}
-
 		if randomDrain {
 			available := make([]uint32, count)
 			for r := uint32(0); r < count; r++ {
 				available[r] = r + 1
 			}
 
-			for r := uint32(0); r < count; r++ {
-				index := rand.Uint32() % (count - r)
-				tmp := available[index]
-				available[index] = available[count-r-1]
-				seqSet := &imap.SeqSet{}
-				seqSet.AddNum(tmp)
-				list = append(list, seqSet)
+			if generateIntervals {
+				const maxIntervalRange = uint32(40)
+				for len(available) > 0 {
+					intervalRange := rand.Uint32() % maxIntervalRange
```

Comment:
`rand.Int()` or `rand.Int31n()` instead of modulo.

Reply:
`maxIntervalRange` is `40` here, so `rand.Uint32() % maxIntervalRange` returns a `uint32` in range `[0, 40)`.

We could equivalently do `uint32(rand.Intn(40))` here but imo the way @LBeernaertProton did it is fine.

Reply:
`rand() %` is a bad habit, but if you're OK, I'm fine.